metadata-ingestion/src/datahub/ingestion/source/lookml.py

```diff
@@ -634,7 +660,10 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
 
                 logger.debug(f"Attempting to load view file: {include}")
                 looker_viewfile = viewfile_loader.load_viewfile(
-                    include, model.connection, self.reporter
+                    path=include,
+                    connection=model.connection,
+                    reporter=self.reporter,
+                    update_cache=True,
```

Comment:
Question about the approach here - I would imagine that the "loading file from filesystem" cache should be separate from the markers of "have we produced a workunit for this yet"

It seems that using the cache exclusively for tracking if the workunit has been produced will result in us reading some files from the filesystem multiple times

Reply:
>  the "loading file from filesystem" cache should be separate from the markers of "have we produced a workunit for this yet"

I totally agree with you! I think it would be better to keep around a separate cache in the `LookMLSource` that tracks "have we produced a workunit for this yet".

The only reason I didn't go with that approach in this PR was that I felt it was a more invasive change. If you want me to do the work to separate those two concerns in this PR, I'd be happy to!

Reply:
I think the "have we produced a workunit for this yet" cache can be a local variable in the `get_workunits` method, which should prevent it from being too invasive. That way, the "loading file from filesystem" cache behavior would also remain unchanged.

Would be great if you could make that edit!

Reply:
> I think the "have we produced a workunit for this yet" cache can be a local variable in the get_workunits method

ah yeah, good point. Ok for sure, I can make that change here. I like that a lot better than the current state of this PR 😀